dm/docs/RFCS/20220124_enhance_task_manageability.md

```diff
@@ -0,0 +1,127 @@
+# Proposal: Enhance task manageability
+
+- Author(s): [ehco](https://github.com/Ehco1996)
+- Last updated: 2022-01-22
+
+## Background
+
+The main purpose of this change is to address the root cause of problems like [#3771](https://github.com/pingcap/tiflow/issues/3771), which are caused by the fact that DM Task itself does not distinguish between dynamic configuration and static resources, making it impossible for users to intuitively manage their tasks. For this reasons, we will attempt to redesign the state machine of DM Task and optimising the DMCTL interaction interface to provide a better user experience.
```

Comment:
```suggestion
The main purpose of this change is to address the root cause of problems like [#3771](https://github.com/pingcap/tiflow/issues/3771), which are caused by the fact that DM task itself does not distinguish between dynamic configuration and static resources, making it impossible for users to intuitively manage their tasks. For this reasons, we attempt to redesign the state machine of DM task and optimize the dmctl interface to provide a better user experience.
```
I think dmctl and others doesn't need to be capitalized

and what's "distinguish between dynamic configuration and static resources"? could you explain it in comment so we can find a better way to express?

Reply:
The main point I was trying to make was the previous confusion between the create-task and start-task operations.
Creating a task should just create a static task configuration, it shouldn't go on to start the task

It is because `dmctl start-task` mixes up the create and start operations that we need an extra pair of operations (pause/resume)